src/iota/view.rs

```diff
@@ -157,17 +157,11 @@ impl View {
         let mut cursor_status = buffer.get_mark_display_coords(self.cursor).unwrap_or((0,0));
         cursor_status = (cursor_status.0 + 1, cursor_status.1 + 1);
         let status_text = format!("{} ({}, {})", buffer_status, cursor_status.0, cursor_status.1).into_bytes();
-        let status_text_len = status_text.len();
         let width = self.get_width();
         let height = self.get_height() - 1;
 
-
-        for index in 0..width {
-            let mut ch: char = ' ';
-            if index < status_text_len {
-                ch = status_text[index] as char;
-            }
-            self.uibuf.update_cell(index, height, ch, CharColor::Black, CharColor::Blue);
+        for (index, ch) in status_text.iter().enumerate().take(width) {
+            self.uibuf.update_cell(index, height, *ch as char, CharColor::Black, CharColor::Blue);         
         }
```

Comment:
This actually changes things slightly...

Basically, I need the status bar to go the full width of the terminal window. In the existing code, I iterate from 0 to the width of the window, and draw the status text for the first portion of that, and for the rest just draw a blank status bar.

With the change you have here, the status bar will only every be as wide as the status text.

I'm not sure if its possible to use iterators and still go the full width of the window even when the status text is much shorter?